### PR TITLE
Replace `validate_bool` with `Boolean` data type

### PR DESCRIPTION
**Pull request by alexjfisher**

### CODE REVIEW

File: manifests/client.pp [hunk 1]
```diff
@@ -46,17 +46,14 @@
 # Copyright (C) 2012 Mike Arnold, unless otherwise noted.
 #
 class snmp::client (
-  $snmp_config        = $snmp::params::snmp_config,
-  $ensure             = $snmp::params::ensure,
-  $autoupgrade        = $snmp::params::safe_autoupgrade,
-  $package_name       = $snmp::params::client_package_name
+  $snmp_config         = $snmp::params::snmp_config,
+  $ensure              = $snmp::params::ensure,
```
Comment (ekohl) [inline manifests/client.pp]: Want to make this an `Enum['present', 'absent']` while you're at it? It's fine to do that in another PR.

File: manifests/client.pp [hunk 2]
```diff
@@ -46,17 +46,14 @@
 # Copyright (C) 2012 Mike Arnold, unless otherwise noted.
 #
 class snmp::client (
-  $snmp_config        = $snmp::params::snmp_config,
-  $ensure             = $snmp::params::ensure,
```
Comment (alexjfisher) [inline manifests/client.pp]: @ekohl Oops.  I seem to have fat fingered here and nuked a few lines I didn't mean to!

File: manifests/client.pp (hunk 1 above)
Comment (alexjfisher) [reply manifests/client.pp]: Considering I randomly deleted a few lines just removing `validate_bool` *and* you approved it, probably best we keep with small PRs (and I get more sleep!) ;)
